### PR TITLE
Add --print0 to `history list`

### PR DESCRIPTION
**Pull request by offbyone**
Adding this enables tools like fzf to handle multiline commands, in the
same way that `find -print0 | fzf --read0` does.

I'm happy to add documentation and changelog entries to this as suits you; just
let me know where to put them.

### CODE REVIEW

File: atuin/src/command/client/history.rs [hunk 1]
```diff
@@ -126,8 +136,16 @@ pub fn print_list(h: &[History], list_mode: ListMode, format: Option<&str>, reve
         Box::new(h.iter()) as Box<dyn Iterator<Item = &History>>
     };
 
+    let entry_terminator = if print0 { "\0" } else { "\n" };
```
Comment (conradludgate) [inline atuin/src/command/client/history.rs:139]: No newline with nulls seems weird. Is that actually how the other tools work?

Comment (offbyone) [reply atuin/src/command/client/history.rs:139]: Yes; the point of this is that if the string might contain newlines -- say, if the command does -- then this allows linewise processors to know where the boundaries of the command are.

The common pattern here is `find -print0 | xargs -0` which allows for filenames with all sorts of bonkers delimiter-like characters.

Comment (offbyone) [reply atuin/src/command/client/history.rs:139]: I _do_ have one rust-ish question on this, though; I'm used to IO that flushes on newlines, which is why I am only calling `flush` when the delimiter is null. However, I'm not entirely sure if the output flushing for `write!` will behave that way, or if I'd need to flush when calling `write!(...something with a newline)` manually.

Comment (conradludgate) [reply atuin/src/command/client/history.rs:139]: You are correct. write for std::io::stdout will flush on  newlines automatically. What you are doing here is equivalent

Comment (offbyone) [reply atuin/src/command/client/history.rs:139]: Perfect. Well, let me know if you want any other details around this change; I'd love to get multiline history search in my shell again, and this is a key step)

**Review by ellie — APPROVED** [review]
This is great, thank you!

Seeing as this is your first time contributing, if you would like a holographic contributors-only Atuin sticker, then please fill out this [form!](https://notionforms.io/forms/contributors-stickers)

We do also have a [Discord](https://discord.gg/Fq8bJSKPHh) if you'd like to ask any questions, or just fancy hanging out!

(also - nice work on your mastodon instance!)

File: atuin/src/command/client/history.rs (hunk 1 above)
Comment (ellie) [reply atuin/src/command/client/history.rs:139]: > I'd love to get multiline history search in my shell again

Mind if I ask where we're lacking in that regard?

Comment (offbyone) [reply atuin/src/command/client/history.rs:139]: Honestly... I don't like the built-in search. I find the search tools fzf offers more enjoyable from my side, so finding a way to get atuin and fzf to play nice is my current goal. 

So, I've been working on history search functions that combine the output of `atuin`, a bit of null-terminated `gawk` scripting, and `fzf` to do the search.

**Review by conradludgate — APPROVED** [review]

**Comment by offbyone** [comment]
> (also - nice work on your mastodon instance!)

Thanks on that last! It kind of fell into my life; I didn't start it, but I am keeping it afloat.

**Comment by offbyone** [comment]
Will this be in 17.0? When (roughly) will that be released? I'm gonna be happy to stop running a homebuild version :D

**Comment by ellie** [comment]
> Will this be in 17.0? 

Absolutely!

> When (roughly) will that be released?

Sometime in the next couple of weeks, time depending. There were some features (sync changes) I wanted to try and get in, but between myself + Conrad having some time off lately that didn't happen

**Pull request merged**